compiler/rustc_borrowck/src/diagnostics/mutability_errors.rs

```diff
@@ -710,6 +705,63 @@ impl<'a, 'tcx> MirBorrowckCtxt<'a, 'tcx> {
         )
     }
 
+    fn construct_suggestion(&self, err: &mut DiagnosticBuilder<'_, ErrorGuaranteed>, local: Local) {
+        let local_decl = &self.body.local_decls[local];
+        debug!("local_decl: {:?}", local_decl);
+        let pat_span = match *local_decl.local_info() {
+            LocalInfo::User(BindingForm::Var(mir::VarBindingForm {
+                binding_mode: ty::BindingMode::BindByValue(Mutability::Not),
+                opt_ty_info: _,
+                opt_match_place: _,
+                pat_span,
+            })) => pat_span,
+            _ => local_decl.source_info.span,
+        };
+
+        struct BindingFinder {
+            span: Span,
+            hir_id: Option<hir::HirId>,
+        }
+
+        impl<'tcx> Visitor<'tcx> for BindingFinder {
+            fn visit_stmt(&mut self, s: &'tcx hir::Stmt<'tcx>) {
+                if let hir::StmtKind::Local(local) = s.kind {
+                    if local.pat.span == self.span {
+                        self.hir_id = Some(local.hir_id);
+                    }
+                }
+                hir::intravisit::walk_stmt(self, s);
+            }
+        }
+
+        let hir_map = self.infcx.tcx.hir();
+        let def_id = self.body.source.def_id();
+        let hir_id = if let Some(local_def_id) = def_id.as_local() && let Some(body_id) = hir_map.maybe_body_owned_by(local_def_id) {
+                let body = hir_map.body(body_id);
+                let mut v = BindingFinder {
+                    span: pat_span,
+                    hir_id: None,
+                };
+                v.visit_body(body);
+                v.hir_id
+            } else {
+                None
+            };
+
+        // suggestion for issue: #114896
+        // let &b = a;
+        // b.make_ascii_uppercase();
+        if let Some(hir_id) = hir_id && let Some(hir::Node::Local(hir::Local {pat: hir::Pat{kind: hir::PatKind::Ref(_, _), ..}, ..})) = hir_map.find(hir_id) {
+                err.span_suggestion(pat_span, "consider changing this to be mutable", format!("&(mut {})", self.infcx.tcx.sess.source_map().span_to_snippet(local_decl.source_info.span).unwrap()), Applicability::MachineApplicable,);
```

Comment:
nit: likewise with the inside of this condition

Reply:
> nit: likewise with the inside of this condition

done